sdks/python/test-suites/dataflow/common.gradle

```diff
@@ -158,19 +168,47 @@ task validatesRunnerStreamingTests {
   def dataflowWorkerJar = project(":runners:google-cloud-dataflow-java:worker").shadowJar.archivePath
 
   // TODO(BEAM-3544,BEAM-5025): Disable tests with 'sickbay-streaming' tag.
-  def testOpts = basicTestOpts + ["--attr=ValidatesRunner,!sickbay-streaming"]
-  def argMap = ["test_opts": testOpts,
+  // Execute tests with xdists
+  doFirst {
+    def argMap = [
+                "test_opts": basicPytestOpts + ["--numprocesses=8"],
                 "streaming": "true",
                 "sdk_location": files(configurations.distTarBall.files).singleFile,
                 "worker_jar": dataflowWorkerJar,
-                "suite": "validatesRunnerStreamingTests-df${pythonVersionSuffix}"]
-  if (project.hasProperty('useRunnerV2')) {
-    argMap.put("runner_v2", "true")
-    // KMS is not supported for streaming engine.
-    argMap.put("kms_key_name", "\"\"")
+                "suite": "validatesRunnerStreamingTests-df${pythonVersionSuffix}-xdist",
+                "pytest": true, // TODO(BEAM-3713): Remove this once nose is removed
+                "collect": "it_validatesrunner and not no_sickbay_streaming"
+                ]
+    if (project.hasProperty('useRunnerV2')) {
+      argMap.put("runner_v2", "true")
+      // KMS is not supported for streaming engine.
+      argMap.put("kms_key_name", "\"\"")
+    }
+
+    def cmdArgs = mapToArgString(argMap)
+    exec {
+      executable 'sh'
+      args '-c', ". ${envdir}/bin/activate && ${runScriptsDir}/run_integration_test.sh $cmdArgs"
+    }
   }
 
+  // Execute tests that fail with xdists
   doLast {
+    def tests = ["apache_beam/runners/dataflow/dataflow_exercise_streaming_metrics_pipeline_test.py"]
```

Comment:
This test fails with xdist so it runs for a single process in pytest

Reply:
Please use the `no_xdist` marker instead (specified in "collect").

Reply:
Thanks, solved!